### ISSUE TITLE
segmentBins() prints NA, leading to failures

### ISSUE THREAD

**Issue by argymeg** [ccagc/QDNAseq#49]
Following the tutorial steps with the example data provided works fine. When I substitute it with my own data, however, I get this after running `segmentBins()`, instead of my sample name:
```
Performing segmentation:
NA
```
When running with Rscript, the script exits after subsequently running `normalizeSegmentedBins()`, with the following error:

```
Error in callNextMethod(.Object, assayData = assayData, phenoData = phenoData,  :
  could not find function "callNextMethod"
Calls: normalizeSegmentedBins ... makeCgh -> new -> initialize -> initialize -> .local
Execution halted
```

When running with R CMD BATCH, however (or running the commands by hand), the script finishes succesfully and produces results, outwardly as expected.

All of this happens with:
- R 3.4.3, QDNAseq 1.14.0
- R 3.4.1, QDNAseq 1.12.0
- R 3.3.3, QDNAseq 1.10.0

OTOH, R 3.2.4 with QDNAseq 1.6.1 works as expected.

I am attaching a simplified version of my script, that produces the results described above.

[qdnaseq_script_simplified.R.txt](https://github.com/ccagc/QDNAseq/files/1790225/qdnaseq_script_simplified.R.txt)

**Comment by ilarischeinin** [comment]
I think there might be two unrelated things here.

1. What's the name of your bam file? And what's the output from `sampleNames(readCounts)`?

2. If you add this line to your script, does it work with Rscript as well?

```r
library(methods)
```

**Comment by argymeg** [comment]
It appears you are right.

1. My BAM file name is in the format "AB1234-XYZ_final.bam" - I did rename it to exactly that and can confirm the result is the same. The output of `sampleNames(readCounts)` (I had to do `library("Biobase")` for it to work) is `[1] "AB1234-XYZ_final"`. However, if I rename it to something simpler like "Sample_1.bam", it works as expected.

2. Yes, it does, thank you!

**Comment by ilarischeinin** [comment]
Thanks. I don't know why, but the list of packages `R` and `Rscript` load by default are different. That's why with `R` the script works without it, but with `Rscript` does need the `library("methods")` call. (This will [change](https://twitter.com/GaborCsardi/status/949722120375988224) at some point.)

So, I think the `NA` threw you off a bit there, as it seemed it was connected to the error message you saw. As that's not the case, can I close the issue? Or is there still something wrong on how samples are named after the bam files? To be clear, the output from `segmentBins()` should be:
```
Performing segmentation:
<name of sample 1>
<name of sample 2>
....
```

**Comment by argymeg** [comment]
Right, that makes sense - I assumed packages that are part of the R base distribution would always be loaded by default. Thanks for clearing that up!

I think it might still be worth investigating why certain sample names can apparently break the output of `segmentBins()`, but it doesn't seem to affect the actual results, so that's up to you.

**Comment by argymeg** [comment]
FWIW, I did another quick test and it seems it might be related to the presence of dashes in the sample name. 
Sample_1.bam gives:
```
Performing segmentation:
    Segmenting: Sample_1 (1 of 1) ...
```
But Sample-1.bam gives:
```
Performing segmentation:
NA
```

**Comment by ilarischeinin** [comment]
Thanks for investigating!

This might be something that happens within the DNAcopy package (which performs the segmentation). It could of course be circumvented within DNAseq, but implementing a fix is probably not a very high priority.

Anyways, thanks for reporting!

**Comment by mmterpstra** [comment]
pls consider truncating the '-' in samplenames/files to '.' this should work.
